packages/web-react/src/components/TextArea/TextArea.tsx

```diff
@@ -1,9 +1,30 @@
-import React from 'react';
+import React, { forwardRef, useRef, RefObject, ForwardedRef } from 'react';
 import { SpiritTextAreaProps } from '../../types';
 import { TextFieldBase } from '../TextFieldBase';
+import { useAdjustHeight } from './useAdjustHeight';
 
-export const TextArea = (props: SpiritTextAreaProps): JSX.Element => {
-  return <TextFieldBase isMultiline {...props} />;
+/* We need an exception for components exported with forwardRef */
+/* eslint no-underscore-dangle: ['error', { allow: ['_TextArea'] }] */
+const _TextArea = (props: SpiritTextAreaProps, ref: ForwardedRef<HTMLTextAreaElement>): JSX.Element => {
+  const { onInput, isAutoResizing, autoResizingMaxHeight = 400, ...restProps } = props;
+  const elementReference = useRef(ref);
+  const { onInput: onInputHandler } = useAdjustHeight({
+    elementReference,
+    onInput,
+    isAutoResizing,
+    maxHeight: autoResizingMaxHeight,
+  });
+
+  return (
+    <TextFieldBase
+      isMultiline
+      ref={elementReference as RefObject<HTMLTextAreaElement>}
+      onInput={onInputHandler}
```

Comment:
Is there a reason why you are attaching the event to the `oninput` instead of `onchange`? I am just curious because I was expecting an attachment to the `onChange`.

Reply:
@crishpeen just used `onInput` so I used it too. When the `onChange` event fires, we also carry `onBlur` and onFocus, while onInput does not. 

* input events fire on any value changes
* change events fire on direct user-initiated value changes

Otherwise they are basically the same

Reply:
Ok, thanks for the explanation.